File: app/src/test/java/android/emu6502/LabelsTest.java
package android.emu6502;

import android.emu6502.instructions.Symbols;

import org.junit.Test;

import java.util.Collections;

import static org.hamcrest.CoreMatchers.equalTo;
import static org.junit.Assert.assertThat;
import static org.mockito.Mockito.mock;

public class LabelsTest {
  private final Symbols symbols = new Symbols();
  private final Assembler assembler = new Assembler(new Memory(mock(Display.class)), symbols);
  private final Labels labels = new Labels(assembler, symbols);

  @Test public void testAddLabel() {
    labels.indexLines(Collections.singletonList("test:"));
    assertThat(labels.get("test"), equalTo(assembler.getDefaultCodePC()));
  }
}

File: app/src/test/kotlin/android/emu6502/LabelsTest.kt
package android.emu6502

import android.emu6502.instructions.Symbols
import org.hamcrest.CoreMatchers.equalTo
import org.junit.Assert.assertThat
import org.junit.Test
import org.mockito.Mockito.mock

class LabelsTest {
  private val symbols = Symbols()
  private val assembler = Assembler(Memory(mock(Display::class.java)), symbols)
  private val labels = Labels(assembler, symbols)

  @Test fun testAddLabel() {
    labels.indexLines(listOf("test:"))
    assertThat(labels["test"], equalTo(assembler.defaultCodePC))
  }
}
